bundles/org.openhab.binding.hue/src/main/java/org/openhab/binding/hue/internal/handler/Clip2ThingHandler.java

```diff
@@ -0,0 +1,1195 @@
+        logger.debug("{} -> dispose()", resourceId);
+        disposing = true;
+        alertResetTask = cancelTask(alertResetTask, true);
+        dynamicsResetTask = cancelTask(dynamicsResetTask, true);
+        updateDependenciesTask = cancelTask(updateDependenciesTask, true);
+        updateServiceContributorsTask = cancelTask(updateServiceContributorsTask, true);
+        legacyLinkedChannelUIDs.clear();
+        sceneContributorsCache.clear();
+        sceneResourceIds.clear();
+        supportedChannelIdSet.clear();
+        commandResourceIds.clear();
+        serviceContributorsCache.clear();
+        controlIds.clear();
+    }
+
+    /**
+     * Get the bridge handler.
+     *
+     * @throws AssetNotLoadedException if the handler does not exist.
+     */
+    private Clip2BridgeHandler getBridgeHandler() throws AssetNotLoadedException {
+        Bridge bridge = getBridge();
+        if (Objects.nonNull(bridge)) {
+            BridgeHandler handler = bridge.getHandler();
+            if (handler instanceof Clip2BridgeHandler) {
+                return (Clip2BridgeHandler) handler;
+            }
+        }
+        throw new AssetNotLoadedException("Bridge handler missing");
+    }
+
+    /**
+     * Do a double lookup to get the cached resource that matches the given ResourceType.
+     *
+     * @param resourceType the type to search for.
+     * @return the Resource, or null if not found.
+     */
+    private @Nullable Resource getCachedResource(ResourceType resourceType) {
+        String commandResourceId = commandResourceIds.get(resourceType);
+        return Objects.nonNull(commandResourceId) ? serviceContributorsCache.get(commandResourceId) : null;
+    }
+
+    /**
+     * Return a ResourceReference to this handler's resource.
+     *
+     * @return a ResourceReference instance.
+     */
+    public ResourceReference getResourceReference() {
+        return new ResourceReference().setId(resourceId).setType(thisResource.getType());
+    }
+
+    /**
+     * Register the 'DynamicsAction' service.
+     */
+    @Override
+    public Collection<Class<? extends ThingHandlerService>> getServices() {
+        return Set.of(DynamicsActions.class);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command commandParam) {
+        if (RefreshType.REFRESH.equals(commandParam)) {
+            if ((thing.getStatus() == ThingStatus.ONLINE) && updateDependenciesDone) {
+                Future<?> task = updateServiceContributorsTask;
+                if (Objects.isNull(task) || !task.isDone()) {
+                    cancelTask(updateServiceContributorsTask, false);
+                    updateServiceContributorsTask = scheduler.schedule(() -> {
+                        try {
+                            updateServiceContributors();
+                        } catch (ApiException | AssetNotLoadedException e) {
+                            // exceptions will not occur here since thing is already online
```

Comment:
I guess debug log of exception message would then make sense?

Reply:
Ok. Done.